src/views/PancakeSquad/components/Header/SaleProgress.tsx

```diff
@@ -12,13 +12,28 @@ type PreEventProps = {
   totalSupplyMinted: number
 }
 
-const SaleProgressTextMapping: Record<SaleStatusEnum, string> = {
-  [SaleStatusEnum.Pending]: '',
-  [SaleStatusEnum.Premint]: '',
-  [SaleStatusEnum.Presale]: '%remaining% of %total% remaining',
-  [SaleStatusEnum.Sale]: '%remaining% of %total% remaining',
-  [SaleStatusEnum.DrawingRandomness]: 'Randomizing NFT allocation with Chainlink',
-  [SaleStatusEnum.Claim]: '%remaining% of %total% minted',
+const saleProgressTextMapping = (t: ContextApi['t'], saleStatus: SaleStatusEnum, remaining: string, total: string) => {
+  const data = {
+    remaining,
+    total,
+  }
+
+  switch (saleStatus) {
+    case SaleStatusEnum.Pending:
+      return t('', data)
+    case SaleStatusEnum.Premint:
+      return t('', data)
```

Comment:
No need to wrap this in a translation function.

Reply:
Done